cairo_programs/relocate_segments.cairo

```diff
@@ -0,0 +1,44 @@
+from starkware.cairo.common.alloc import alloc
+
+func relocate_segment(src_ptr : felt*, dest_ptr : felt*):
+    %{ 
+        # TEST
+        memory.add_relocation_rule(src_ptr=ids.src_ptr, dest_ptr=ids.dest_ptr) 
+    %}
+
+    # Add a verifier side assert that src_ptr and dest_ptr are indeed equal.
+    assert src_ptr = dest_ptr
+    return ()
+end
+
+
+func main():
+    alloc_locals
+    # Create temporary_array in a temporary segment
+    local temporary_array : felt*
+    %{
+        ids.temporary_array = segments.add_temp_segment()
+    %}
```

Comment:
We seem to be missing the `#TEST` comment, will this use the builtin hint processor?

Reply:
[Done](https://github.com/lambdaclass/cairo-rs-py/pull/112/commits/6def182999be5b3566aeb62934613c8a97b648e2)